### PR TITLE
Group Golf Assignment 1

### PR DESCRIPTION
**Pull request by Archit0303**

### CODE REVIEW

File: two_tower.py [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+
+        embeddings = self.model.forward_user(user_tensor).detach().numpy().astype(np.float32)
+        if len(embeddings) != len(user_tensor) or embeddings.shape[1] > 64:
+            logging.error("Mismatch in embeddings and tensor length or embedding size exceeds 64")
+            return np.array([])
+
+        return embeddings
+
+# Initialize ModelWrapper with an empty path to return a dummy model for testing
+model_wrapper = ModelWrapper()
```
Comment (kennethgoodman) [inline two_tower.py:89]: pass the model path?

File: two_tower_train.py
Comment (kennethgoodman) [inline two_tower_train.py:1]: you need to upload this

File: CollaberativeFilteredSimilarUsersGenerator.py [hunk 2]
```diff
@@ -0,0 +1,15 @@
+
+from src.recommendation_system.recommendation_flow.candidate_generators.AbstractGenerator import AbstractGenerator
+from src.data_structures.user_based_recommender.golf.UserBasedRecommender import UserBasedRecommender
+
+
+class CollaberativeFilteredSimilarUsersGenerator(AbstractGenerator):
+    def _get_content_ids(self, user_id, limit, offset, seed, starting_point):
+
+        recommender = UserBasedRecommender()
```
Comment (kennethgoodman) [inline CollaberativeFilteredSimilarUsersGenerator.py:9]: where is the implementation for this?

File: YourChoiceGenerator.py [hunk 3]
```diff
@@ -0,0 +1,14 @@
+
+from src.recommendation_system.recommendation_flow.candidate_generators.AbstractGenerator import AbstractGenerator
+import numpy as np
+
+#random generator
+
+class YourChoiceGenerator(AbstractGenerator):
+    def _get_content_ids(self, _, limit, offset, _seed, starting_point):
+        content_id = np.random.randint(low=30001,high=100000,size=(1000,))
```
Comment (kennethgoodman) [inline YourChoiceGenerator.py:9]: please query from the database, what if content gets deleted?

**Review by kennethgoodman — COMMENTED** [review]
Please address the comments AND edit the files in the code in the right spot, see other PRs if you're unsure

File: two_tower_train.py
Comment (zwht81) [reply two_tower_train.py:1]: I rename it. See the two_tower_train_popular_negative_sampling.py and two_tower_train_random_negative_sampling.py

File: two_tower.py (hunk 1 above)
Comment (YifanSun99) [reply two_tower.py:208]: We pass the model path as a default parameter in the function

File: CollaberativeFilteredSimilarUsersGenerator.py (hunk 2 above)
Comment (Archit0303) [reply CollaberativeFilteredSimilarUsersGenerator.py:9]: uploaded this file just now [UserBasedRecommender.py]

**Comment by kennethgoodman** [comment]
You need to edit the files already existing in the DB not upload new ones

**Comment by Archit0303** [comment]
> You need to edit the files already existing in the DB not upload new ones

I had forgotten to upload some files so uploaded those.

**Comment by kennethgoodman** [comment]
Please see other PRs, it looks like your just uploading new files?

**Comment by Archit0303** [comment]
> Please see other PRs, it looks like your just uploading new files?

I see those PRs and would want to do the same for our team too but my colleague has made the two tower train file as separate file which had to be uploaded. Also some other files I accidently merged the original one's back to your repo. If it's fine can I delete this fork and pull request and do the merge again with all the correct uploaded files?

**Comment by kennethgoodman** [comment]
You're free to do another PR if that makes the most sense

**Comment by Archit0303** [comment]
> You're free to do another PR if that makes the most sense

just commited all the changes to this repo

**Comment by kennethgoodman** [comment]
thank you, Please delete the other files now

**Comment by Archit0303** [comment]
> thank you, Please delete the other files now

thank you professor for your patience and support, I have deleted the other files outside the folder now.

**Review by kennethgoodman — APPROVED** [review]

**Pull request merged**